src/test/java/automationpractice/com/pages/ProductDetailsPage.java

```diff
@@ -0,0 +1,251 @@
+            listOfColors.add(color[0]);
+        }
+        return listOfColors;
+    }
+
+    public String changeColorToGivenColorInProductDetailsPage(String colorStartedWithCapitalLetter) {
+        currentColorInProductDetailsPage = colorStartedWithCapitalLetter;
+        ArrayList listOfAvailableColors = getAvailableColorsInProductDetailsPage();
+
+        for (Object listOfAvailableColor : listOfAvailableColors) {
+            if (listOfAvailableColor.toString().equals(colorStartedWithCapitalLetter)) {
+                switch (colorStartedWithCapitalLetter) {
+
+                    case "Orange":
+                        changeColorToOrange.click();
+                        break;
+                    case "White":
+                        changeColorToWhite.click();
+                        break;
+                    case "Black":
+                        changeColorToBlack.click();
+                        break;
+                    case "Green":
+                        changeColorToGreen.click();
+                        break;
+                    case "Yellow":
+                        changeColorToYellow.click();
+                        break;
+                    case "Blue":
+                        changeColorToBlue.click();
+                        break;
+                }
+            }
+        }
+        // TODO: 02.01.2019 default choose first available color
+        return colorStartedWithCapitalLetter;
+    }
+
+    public String getCurrentColorSetUpInProductDetailsPage() {
+        return currentColorInProductDetailsPage;
+    }
+
+    public void clickOnTwitterButton() {
+        clickOnTwitterButton.click();
+    }
+
+    public void clickOnFacebookButton() {
+        clickOnFacebookButton.click();
+    }
+
+    public void clickOnGooglePlusButton() {
+        clickOnGooglePlusButton.click();
+    }
+
+    public void clickOnPinterestButton() {
+        clickOnPinteresButton.click();
+    }
+
+    public void checkoutToTweeterWindow() {
+        waitABit(1000);
+        for (String s : getDriver().getWindowHandles()) {
+            if (!s.equals("2147483649")) {
+                getDriver().switchTo().window(s);
+            }
+        }
+    }
+
+    public void checkoutToFacebookWindow() {
+        waitABit(1000);
+        for (String s : getDriver().getWindowHandles()) {
+            if (!s.equals("2147483649")) {
+                getDriver().switchTo().window(s);
+            }
+        }
+    }
+
+    public void checkoutToGooglePlusWindow() {
+        waitABit(1000);
+        for (String s : getDriver().getWindowHandles()) {
+            if (!s.equals("2147483649")) {
+                getDriver().switchTo().window(s);
+            }
+        }
+    }
+
+    public void checkoutToPinterestWindow() {
+        waitABit(3000);
+        for (String s : getDriver().getWindowHandles()) {
+            if (!s.equals("2147483649")) {
+                getDriver().switchTo().window(s);
+            }
+        }
+    }
+
```

Comment:
Te 4 metody są właściwie identyczne. Powtarzasz ten sam kod.
Zamiast for i if możesz też zrobić tak:

final ArrayList<String> tabs = new ArrayList<>(getDriver().getWindowHandles());
getDriver().switchTo().window(tabs.get(1));